js/lib/beautify.js

```diff
@@ -423,7 +448,11 @@
                 return
             }
 
-            if ((opt.preserve_newlines && current_token.wanted_newline) || force_linewrap) {
+            var shouldPreserveOperatorNewline = (opt.operator_position === OPERATOR_POSITION.preserve_newline) &&
+                in_array(flags.last_text, OP_POSITION_OPERATORS) ||
+                in_array(current_token.text, OP_POSITION_OPERATORS);
+                
+            if (((opt.preserve_newlines || shouldPreserveOperatorNewline) && current_token.wanted_newline) || force_linewrap) {
```

Comment:
... Actually, the more I think about it, if `preserve_newlines = false`, that setting should take precedence for input requested newlines.  That will make the behavior most consistent.  If you turn off preserve_newlines you just turn them _off_.  If you allow them, then you can control where they are (or remove them) for operators.

The line-wrapping behavior is covered in by when the allow_wrap_or_preserve_newline gets called...  I'll need to review the inputs and outputs more carefully for this case.